node-graph/interpreted-executor/src/node_registry.rs

```diff
@@ -452,6 +454,61 @@ static NODE_REGISTRY: &[(NodeIdentifier, NodeConstructor)] = &[
 			}
 		},
 	),
+	(NodeIdentifier::new("graphene_std::memo::CacheNode", &[concrete!("Image")]), |proto_node, stack| {
+		let node_id = proto_node.input.unwrap_node() as usize;
+		use graphene_core::raster::*;
+		if let ConstructionArgs::Nodes(image_args) = proto_node.construction_args {
+			stack.push_fn(move |nodes| {
+				let image = nodes.get(node_id).unwrap();
+				let node: DynAnyNode<_, Image, Image, &Image> = DynAnyNode::new(CacheNode::new());
+
+				let node = image.then(node);
+				node.into_type_erased()
+			})
+		} else {
+			unimplemented!()
+		}
+	}),
+	(NodeIdentifier::new("graphene_core::raster::BlurNode", &[]), |proto_node, stack| {
+		let node_id = proto_node.input.unwrap_node() as usize;
+		use graphene_core::raster::*;
+
+		static EMPTY_IMAGE: ValueNode<Image> = ValueNode::new(Image::empty());
+		if let ConstructionArgs::Nodes(blur_args) = proto_node.construction_args {
+			stack.push_fn(move |nodes| {
+				let pre_node = nodes.get(node_id).unwrap();
+				let radius = nodes.get(blur_args[0] as usize).unwrap();
+				let sigma = nodes.get(blur_args[1] as usize).unwrap();
+				let radius = DowncastBothNode::<_, (), u32>::new(radius);
+				let sigma = DowncastBothNode::<_, (), f64>::new(sigma);
+				let image = DowncastBothNode::<_, Image, &Image>::new(pre_node);
+				// dirty hack: we abuse that the cache node will ignore the input if it is
+				// evaluated a second time
```

Comment:
Could we insert a cache proto-node here temporarily so that it doesn't have to be a document node that the user has to insert?

Reply:
I had some issues because the lambda  u32 -> Color also references the image node and I can't return a reference to data owned by the current function. I think the borrow stack does the heavy lifting in this case but I might need to double  check that